loopd/daemon.go

```diff
@@ -119,38 +136,39 @@ func daemon(config *config, lisCfg *listenerCfg) error {
 	// We'll also create and start an accompanying proxy to serve clients
 	// through REST.
 	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
+	d.restCtxCancel = cancel
 	mux := proxy.NewServeMux(customMarshalerOption)
 	var restHandler http.Handler = mux
-	if config.CORSOrigin != "" {
-		restHandler = allowCORS(restHandler, config.CORSOrigin)
+	if d.cfg.CORSOrigin != "" {
+		restHandler = allowCORS(restHandler, d.cfg.CORSOrigin)
 	}
 	proxyOpts := []grpc.DialOption{
 		grpc.WithInsecure(),
 		grpc.WithDefaultCallOptions(maxMsgRecvSize),
 	}
 	err = looprpc.RegisterSwapClientHandlerFromEndpoint(
-		ctx, mux, config.RPCListen, proxyOpts,
+		ctx, mux, d.cfg.RPCListen, proxyOpts,
 	)
 	if err != nil {
 		return err
 	}
 
-	restListener, err := lisCfg.restListener()
+	d.restListener, err = d.listenerCfg.restListener()
 	if err != nil {
 		return fmt.Errorf("REST proxy unable to listen on %s",
-			config.RESTListen)
+			d.cfg.RESTListen)
 	}
 
 	// A nil listener indicates REST is disabled.
-	if restListener != nil {
+	if d.restListener != nil {
 		log.Infof("Starting REST proxy listener")
 
-		defer restListener.Close()
-		proxy := &http.Server{Handler: restHandler}
+		d.restServer = &http.Server{Handler: restHandler}
 
 		go func() {
-			err := proxy.Serve(restListener)
+			log.Infof("REST proxy listening on %s",
+				d.restListener.Addr())
+			err := d.restServer.Serve(d.restListener)
```

Comment:
Error is just logged, no further stopping?

Reply:
Yes, same here. `Serve` only returns on shutdown of the server.

Reply:
And how about other errors?

Reply:
They will now be sent over the `errChan`.